acestep/handler.py

```
@@ -655,31 +671,46 @@ def _synchronize(self):
-                target_type = torch.device(str(target_device)).type
+            target = torch.device(str(target_device))
+            actual = tensor.device
+            # Compare type first
+            if actual.type != target.type:
+                return False
+            # For CUDA devices, also compare index (None == 0)
+            if actual.type == "cuda":
+                actual_idx = actual.index if actual.index is not None else 0
+                target_idx = target.index if target.index is not None else 0
+                return actual_idx == target_idx
+            return True
```

Ruff (0.15.0)

[warning] Consider moving this statement to an `else` block

(TRY300)